daemon/endpoint.go

```diff
@@ -187,14 +182,14 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 		}
 	}
 
+	ep.RLock()
 	endpointmanager.Insert(ep)
+	ep.RUnlock()
```

Comment:
:( It's unfortunate that we need to lock ep here and in Remove below (and in updateReferences too). It would be nicer if all the fields being used would RLock but that's a ton of work :/

In any case, we should add comments to endpointmanager.Insert, endpointmanager.Remove and endpointmanager.UpdateReferences that you need ep RLocked to use them.

Reply:
> In any case, we should add comments to endpointmanager.Insert, endpointmanager.Remove and endpointmanager.UpdateReferences that you need ep RLocked to use them.

Done